otter/integration/tests/test_convergence.py

```diff
@@ -428,6 +429,130 @@ def _scale_down_after_oobd_non_constrained_param(
+            Delete z of the servers out of band
+            Attempt to scale down by (y) servers
+            Validate end state min_servers
+        """
+        rcs = TestResources()
+
+        min_servers = 5
+        z = 2
+        y = -2
+
+        return self._scale_down_after_oobd_hitting_constraints(
+            rcs, oobd_servers=z, min_servers=min_servers,
+            scale_servers=y,
+            converged_servers=min_servers)
+
+    def _assert_error_status_code(self, result, code, rcs):
+        """
+        Validate that the returned value was a failure with the specified
+        status code.
+        """
+        if not isinstance(result, Failure):
+            self.fail('Unexpectedly, this succeeded when it was '
+                      'expected to fail')
+        elif not result.check(APIError):
+            self.fail('Received {0} instead of expected APIError'.format(
+                      result.type))
+        elif result.value.code != code:
+            self.fail('Expected status code {0} but received {1}'.format(
+                      code, result.value.code))
+        else:
+            return rcs
+
+    def _scale_down_after_oobd_hitting_constraints(
+            self, rcs, min_servers=0, max_servers=25, set_to_servers=None,
+            oobd_servers=0, scale_servers=1, converged_servers=0):
+
+        converged_servers = set_to_servers
+
+        scaling_group_body = create_scaling_group_dict(
+            image_ref=image_ref, flavor_ref=flavor_ref,
+            min_entities=min_servers, max_entities=max_servers
+        )
+
+        self.scaling_group = ScalingGroup(
+            group_config=scaling_group_body,
+            pool=self.pool
+        )
+
+        self.policy_scale = ScalingPolicy(
+            scale_by=scale_servers,
+            scaling_group=self.scaling_group
+        )
+        d = self.identity.authenticate_user(
+            rcs,
+            resources={
+                "otter": ("autoscale", "http://localhost:9000/v1.0/{0}"),
+                "nova": ("cloudServersOpenStack",),
+            },
+            region=region
+        ).addCallback(self.scaling_group.start, self)
+
+        if set_to_servers is not None:
+            self.policy_set = ScalingPolicy(
+                set_to=set_to_servers,
+                scaling_group=self.scaling_group
+            )
+            (d.addCallback(self.policy_set.start, self)
+             .addCallback(self.policy_set.execute))
+
+        (d.addCallback(
+            self.scaling_group.wait_for_N_servers,
+            min_servers if set_to_servers is None else set_to_servers,
+            timeout=120)
+         .addCallback(self.scaling_group.get_scaling_group_state)
+         .addCallback(self._choose_random_servers, oobd_servers)
+         .addCallback(self._delete_those_servers, rcs)
+         # The execution of the policy triggers convergence
+         .addCallback(self.policy_scale.start, self)
+          .addCallback(self.policy_scale.execute)
```

Comment:
Non-blocking OCD nitpick - this line doesn't line up.  Actually I'm surprised the linter isn't complaining, but if the linter doesn't complain, I'm not going to block it.  :)